client/service/src/task_manager/mod.rs

```diff
@@ -294,9 +310,27 @@ impl TaskManager {
 			let mut t1 = self.essential_failed_rx.next().fuse();
 			let mut t2 = self.on_exit.clone().fuse();
 
-			futures::select! {
```

Comment:
Why is this whole function not async? Instead of returning a boxed Future?

Reply:
Fair point!! I think I'm used to do that because you can have async fn in traits.

But it still doesn't work:

```
error[E0733]: recursion in an `async fn` requires boxing
   --> client/service/src/task_manager/mod.rs:288:40
    |
288 |     pub async fn clean_shutdown(mut self) {
    |                                           ^ recursive `async fn`
    |
    = note: a recursive `async fn` must be 
```

I isolated the suspect. It seems those 2 lines make the function "recursive":

```
		join_all(children_shutdowns).await;
		completion_future.await;
```

Reply:
Did you remove the internal `async move`?

Reply:
yes obviously XD